service/handlers.go

```diff
@@ -201,3 +201,17 @@ func (rh *RouteHandler) convertToWebhookGenericObject(genericObject client.Gener
 		Config: config,
 	}, nil
 }
+
+func (rh *RouteHandler) isUniqueName(webhookName string, projectID string) (int, error) {
+	apiClient, err := rh.ClientFactory.GetClient(projectID)
```

Comment:
pass the apiClient in as opposed to creating it here. You should call the clientFactory once and pass the same client to this and validate